### PR TITLE
Feature summary

### LINKED ISSUE #1: Implement metricq-summary in Python
Having a python variant of `metricq-summary` would be great for environments where it is hard to compile software. Yes that is weird. Looking at you taurus-ml.

But it could be a good example for a drain. Anyway... maybe this will be wontfix. We'll see.

### PR DESCRIPTION
**Pull request by Daddelhai**
Added summary.py with support for multiple metrics

fixes #1

### CODE REVIEW

File: metricq_tools/summary.py [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+
+        stdout, stderr = await proc.communicate()
+
+        click.echo(f'{self.command!r} exited with {proc.returncode}')
+        if stdout:
+            click.echo(stdout.decode())
+        if stderr:
+            click.echo(stderr.decode())
+
+    async def _on_data_message(self, message: aio_pika.IncomingMessage):
```
Comment (phijor) [inline metricq_tools/summary.py:124]: Please add a comment why it is necessary to override `_on_data_message`.

I guess this is here to keep track of the sources that produce these metrics, right?

**Review by phijor — CHANGES_REQUESTED** [review]
Very nice work, thank you!

Just a few minor comments from my side, but looks good otherwise.

**Comment by Daddelhai** [comment]
Moved to #3

File: metricq_tools/summary.py (hunk 1 above)
Comment (Daddelhai) [reply metricq_tools/summary.py]: wasn't necessary anymore and has been removed

File: metricq_tools/summary.py [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+        await self.stop()
+        self.print_histograms()
+
+    async def run_cmd(self):
+        click.echo(f'running... {self.command!r}')
+
+        proc = await asyncio.create_subprocess_shell(
+            self.command,
+            stdout=asyncio.subprocess.PIPE,
+            stderr=asyncio.subprocess.PIPE)
```
[the remark this answers is not in the document]
Comment (Daddelhai) [reply metricq_tools/summary.py]: I've formatted it correctly

File: metricq_tools/summary.py [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+
+
+        super().__init__(*args, client_version=client_version, **kwargs)
+
+    async def connect(self):
+        await super().connect()
+
+        click.echo(
+            click.style(
+                f"Inspecting the metric '{self._metrics}'...",
```
[the remark this answers is not in the document]
Comment (Daddelhai) [reply metricq_tools/summary.py:88]: done

File: metricq_tools/summary.py [hunk 4]
```diff
@@ -0,0 +1,339 @@
+#!/usr/bin/env python3
+# Copyright (c) 2019, ZIH, Technische Universitaet Dresden, Federal Republic of Germany
```
[the remark this answers is not in the document]
Comment (Daddelhai) [reply metricq_tools/summary.py:2]: done

**Pull request closed without merging**